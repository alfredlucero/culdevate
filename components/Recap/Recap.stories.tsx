import React from "react";
import { storiesOf } from "@storybook/react";
import StoryFrame from "../__common__/StoryFrame";
import Recap, { RecapProps } from "./index";

const allRecapSampleProps: RecapProps[] = [
  {
    title: "Summaries and Goals",
    description: "I endeavor to achieve something!",
    type: "objectives",
    startDate: "2019/04/20",
    endDate: null,
  },
  {
    title: "Sandia National Labs",
    description: "Web Developer Intern",
    type: "work",
    startDate: "2016/06/20",
    endDate: "2017/06/10",
  },
  {
    title: "University of California, Los Angeles",
    description: "BS in Computer Science",
    type: "education",
    startDate: "2013/10/01",
    endDate: "2017/6/10",
  },
  {
    title: "Varsity Tennis",
    description: "SJB Tennis all 4 years",
    type: "extracurricular",
    startDate: "2009/09/15",
    endDate: "2013/5/20",
  },
  {
    title: "Terraform",
    description: "Learned how to do infrastructure as code with AWS stuff!",
    type: "skills",
    startDate: "2018/01/10",
    endDate: null,
  },
  {
    title: "Promoted to Software Engineer II",
    description: "After 1 year and many projects, I earned my stripes!",
    type: "achievements",
    startDate: "2018/10/20",
    endDate: null,
  },
  {
    title: "Arm in Arm Award",
    description: "Helped out CGE with Billing Rewrite to steer them back on time.",
    type: "awards",
    startDate: "2019/01/07",
    endDate: "2019/05/20",
  },
  {
    title: "Reference from Sandia Mentors",
    description: "Can check them out on LinkedIn",
    type: "proof",
    startDate: "2016/06/20",
    endDate: "2017/06/05",
  },
  {
    title: "Code Complete Book",
    description: "Finished reading this book!",
    type: "other",
    startDate: "2018/10/12",
    endDate: "2019/07/25",
  },
];

storiesOf("Recap", module).add("All Recap Types", () => (
  <StoryFrame>
    {allRecapSampleProps.map((recapSampleProps, idx) => (
      <Recap {...recapSampleProps} key={idx} />
    ))}
  </StoryFrame>
));
