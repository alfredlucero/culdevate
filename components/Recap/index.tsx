import React from "react";
import styled from "styled-components";
import Icon, { CuldevateIcon, StyledRoundIcon } from "../__common__/Icon";
import { culdevateDefaultTheme } from "../defaultTheme";

export type RecapType =
  | "objectives"
  | "work"
  | "education"
  | "extracurricular"
  | "skills"
  | "achievements"
  | "awards"
  | "proof"
  | "other";

export interface RecapProps {
  title: string;
  description: string;
  type: RecapType;
  startDate: string;
  endDate: string | null;
}

const mapRecapTypeToIcon = (recapType: RecapType): React.ReactNode | null => {
  const recapTypeToIconMap: { [key in RecapType]: CuldevateIcon } = {
    objectives: "target",
    work: "suitcase",
    education: "school",
    extracurricular: "userFriends",
    skills: "levelUp",
    achievements: "trophy",
    awards: "award",
    proof: "pen",
    other: "checkDouble",
  };
  const mappedIcon = recapTypeToIconMap[recapType];

  return mappedIcon ? (
    <StyledRoundIcon>
      <Icon icon={mappedIcon} />
    </StyledRoundIcon>
  ) : null;
};

const StyledRecap = styled.article`
  display: flex;

  & figure {
    padding: 30px;
    flex: 0 0 50px;
    margin: 0;
  }
`;

StyledRecap.defaultProps = {
  theme: culdevateDefaultTheme,
};

const StyledRecapSection = styled.section`
  flex: 1 0 auto;
`;

const StyledRecapHeader = styled.header`
  display: flex;
  justify-content: space-between;

  & time {
    font-size: 2rem;
    color: ${props => props.theme.colors.orange};
  }
`;

StyledRecapHeader.defaultProps = {
  theme: culdevateDefaultTheme,
};

const Recap: React.FC<RecapProps> = ({ title, description, type, startDate, endDate }) => {
  return (
    <StyledRecap>
      <figure>{mapRecapTypeToIcon(type)}</figure>
      <StyledRecapSection>
        <StyledRecapHeader>
          <h2>{title}</h2>
          <h3>
            <time data-testid="recapDateRange">
              <em>
                {startDate} - {endDate !== null ? endDate : "Present"}
              </em>
            </time>
          </h3>
        </StyledRecapHeader>
        <p>{description}</p>
        <footer>Edit</footer>
      </StyledRecapSection>
    </StyledRecap>
  );
};

export default Recap;
