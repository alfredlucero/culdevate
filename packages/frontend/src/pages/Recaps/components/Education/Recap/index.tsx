import React from "react";
import cn from "classnames";
import { format, parseISO } from "date-fns";
import Card from "../../../../../components/Card";
import Text from "../../../../../components/Text";
import Heading from "../../../../../components/Heading";
import Button from "../../../../../components/Button";
import Icon from "../../../../../components/Icon";
import BulletPoint from "../../BulletPoint";
import RecapIcon from "../../RecapIcon";
import { RecapEducation } from "../../../recaps.interface";
import { CommonProps } from "../../../../../components/commonProps";

interface EducationRecapProps extends CommonProps {
  education: RecapEducation;
  onEdit: (e: React.MouseEvent) => void;
  onDelete: (e: React.MouseEvent) => void;
}

const EducationRecap: React.FC<EducationRecapProps> = ({
  education,
  onEdit,
  onDelete,
  className = "",
  testId = "",
  ...passThroughProps
}) => {
  return (
    <Card className={cn("p-4", "flex", className)} testId={testId} {...passThroughProps}>
      <RecapIcon variant="education" className="mr-4" />
      <div className="flex-auto">
        <Heading variant="h4">{education.school}</Heading>
        <Text variant="p">{education.location}</Text>
        <Text variant="p">
          {education.degree}, {education.fieldOfStudy}
        </Text>
        <Text variant="p" italic={true} className="text-gray-600">
          {format(parseISO(education.startDate), "yyyy")} -{" "}
          {education.endDate ? format(parseISO(education.endDate), "yyyy") : "Present"}
          <br />
          {education.grade}
        </Text>
        <ul className="mt-3">
          {education.bulletPoints.map((bulletPoint, index) => (
            <BulletPoint bulletPoint={bulletPoint} key={index} />
          ))}
        </ul>
      </div>
      <div className="w-1/3 text-right">
        <Button id={education._id} type="button" variant="secondary" onClick={onEdit} className="mr-2">
          <Icon variant="editPencil" size="small" className="mr-2" /> Edit
        </Button>
        <Button id={education._id} type="button" variant="danger" onClick={onDelete}>
          <Icon variant="trash" size="small" className="mr-2" />
          Delete
        </Button>
      </div>
    </Card>
  );
};

export default EducationRecap;
