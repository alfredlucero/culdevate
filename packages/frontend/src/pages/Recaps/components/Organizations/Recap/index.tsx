import React from "react";
import cn from "classnames";
import { format } from "date-fns";
import Card from "../../../../../components/Card";
import Text from "../../../../../components/Text";
import Heading from "../../../../../components/Heading";
import Button from "../../../../../components/Button";
import Icon from "../../../../../components/Icon";
import BulletPoint from "../../BulletPoint";
import RecapIcon from "../../RecapIcon";
import { RecapOrganizations } from "../../../recaps.interface";
import { CommonProps } from "../../../../../components/commonProps";

interface OrganizationsRecapProps extends CommonProps {
  organizations: RecapOrganizations;
  onEdit: (e: React.MouseEvent) => void;
  onDelete: (e: React.MouseEvent) => void;
}

const OrganizationsRecap: React.FC<OrganizationsRecapProps> = ({
  organizations,
  onEdit,
  onDelete,
  className = "",
  testId = "",
  ...passThroughProps
}) => {
  return (
    <Card className={cn("p-4", "flex", className)} testId={testId} {...passThroughProps}>
      <RecapIcon variant="organizations" className="mr-4" />
      <div className="flex-auto">
        <Heading variant="h4">{organizations.organizationName}</Heading>
        <Text variant="p">{organizations.location}</Text>
        <Text variant="p">{organizations.positions}</Text>
        <Text variant="p" italic={true} className="text-gray-600">
          {format(organizations.startDate, "yyyy")} -{" "}
          {organizations.endDate ? format(organizations.endDate, "yyyy") : "Present"}
        </Text>
        <ul className="mt-3">
          {organizations.bulletPoints.map((bulletPoint, index) => (
            <BulletPoint bulletPoint={bulletPoint} key={index} />
          ))}
        </ul>
      </div>
      <div className="w-1/3 text-right">
        <Button id={organizations._id} type="button" variant="secondary" onClick={onEdit} className="mr-2">
          <Icon variant="editPencil" size="small" className="mr-2" /> Edit
        </Button>
        <Button id={organizations._id} type="button" variant="danger" onClick={onDelete}>
          <Icon variant="trash" size="small" className="mr-2" />
          Delete
        </Button>
      </div>
    </Card>
  );
};

export default OrganizationsRecap;
