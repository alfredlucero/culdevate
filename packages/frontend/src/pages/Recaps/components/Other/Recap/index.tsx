import React from "react";
import cn from "classnames";
import { format, parseISO } from "date-fns";
import Card from "../../../../../components/Card";
import Text from "../../../../../components/Text";
import Heading from "../../../../../components/Heading";
import Button from "../../../../../components/Button";
import Icon from "../../../../../components/Icon";
import BulletPoint from "../../BulletPoint";
import RecapIcon from "../../RecapIcon";
import { RecapOther } from "../../../recaps.interface";
import { CommonProps } from "../../../../../components/commonProps";

interface OtherRecapProps extends CommonProps {
  other: RecapOther;
  onEdit: (e: React.MouseEvent) => void;
  onDelete: (e: React.MouseEvent) => void;
}

const OtherRecap: React.FC<OtherRecapProps> = ({
  other,
  onEdit,
  onDelete,
  className = "",
  testId = "",
  ...passThroughProps
}) => {
  return (
    <Card className={cn("p-4", "flex", className)} testId={testId} {...passThroughProps}>
      <RecapIcon variant="other" className="mr-4" />
      <div className="flex-auto">
        <Heading variant="h4">{other.title}</Heading>
        <Text variant="p" italic={true} className="text-gray-600">
          {format(parseISO(other.startDate), "MMMM yyyy")} - {format(parseISO(other.endDate), "MMMM yyyy")}
        </Text>
        <ul className="mt-3">
          {other.bulletPoints.map((bulletPoint, index) => (
            <BulletPoint bulletPoint={bulletPoint} key={index} />
          ))}
        </ul>
      </div>
      <div className="w-1/3 text-right">
        <Button id={other._id} type="button" variant="secondary" onClick={onEdit} className="mr-2">
          <Icon variant="editPencil" size="small" className="mr-2" /> Edit
        </Button>
        <Button id={other._id} type="button" variant="danger" onClick={onDelete}>
          <Icon variant="trash" size="small" className="mr-2" />
          Delete
        </Button>
      </div>
    </Card>
  );
};

export default OtherRecap;
