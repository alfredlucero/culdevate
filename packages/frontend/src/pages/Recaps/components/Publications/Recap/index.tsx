import React from "react";
import cn from "classnames";
import { format } from "date-fns";
import Card from "../../../../../components/Card";
import Text from "../../../../../components/Text";
import Heading from "../../../../../components/Heading";
import Button from "../../../../../components/Button";
import Icon from "../../../../../components/Icon";
import Link from "../../../../../components/Link";
import BulletPoint from "../../BulletPoint";
import RecapIcon from "../../RecapIcon";
import { RecapPublications } from "../../../../../interfaces/recaps.interface";
import { CommonProps } from "../../../../../components/commonProps";

interface PublicationsRecapProps extends CommonProps {
  publications: RecapPublications;
  onEdit: (e: React.MouseEvent) => void;
  onDelete: (e: React.MouseEvent) => void;
}

const PublicationsRecap: React.FC<PublicationsRecapProps> = ({
  publications,
  onEdit,
  onDelete,
  className = "",
  testId = "",
  ...passThroughProps
}) => {
  return (
    <Card className={cn("p-4", "flex", className)} testId={testId} {...passThroughProps}>
      <RecapIcon variant="publications" className="mr-4" />
      <div className="flex-auto">
        <Heading variant="h4">{publications.title}</Heading>
        <Text variant="p">By {publications.coauthors}</Text>
        <Text variant="p" italic={true} className="text-gray-600">
          {format(publications.startDate, "MMMM dd, yyyy")}
        </Text>
        <Text variant="p">
          {publications.publisher} ({publications.type})
        </Text>
        {publications.url && (
          <Text variant="p">
            <Link href={publications.url} type="external">{`${publications.url.substr(0, 30)}...`}</Link>
          </Text>
        )}
        <ul className="mt-3">
          {publications.bulletPoints.map((bulletPoint, index) => (
            <BulletPoint bulletPoint={bulletPoint} key={index} />
          ))}
        </ul>
      </div>
      <div className="w-1/3 text-right">
        <Button id={publications._id} type="button" variant="secondary" onClick={onEdit} className="mr-2">
          <Icon variant="editPencil" size="small" className="mr-2" /> Edit
        </Button>
        <Button id={publications._id} type="button" variant="danger" onClick={onDelete}>
          <Icon variant="trash" size="small" className="mr-2" />
          Delete
        </Button>
      </div>
    </Card>
  );
};

export default PublicationsRecap;
