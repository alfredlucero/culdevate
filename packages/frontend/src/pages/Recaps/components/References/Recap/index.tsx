import React from "react";
import cn from "classnames";
import Card from "../../../../../components/Card";
import Text from "../../../../../components/Text";
import Heading from "../../../../../components/Heading";
import Button from "../../../../../components/Button";
import Icon from "../../../../../components/Icon";
import BulletPoint from "../../BulletPoint";
import RecapIcon from "../../RecapIcon";
import { RecapReferences } from "../../../recaps.interface";
import { CommonProps } from "../../../../../components/commonProps";

interface ReferencesRecapProps extends CommonProps {
  references: RecapReferences;
  onEdit: (e: React.MouseEvent) => void;
  onDelete: (e: React.MouseEvent) => void;
}

const ReferencesRecap: React.FC<ReferencesRecapProps> = ({
  references,
  onEdit,
  onDelete,
  className = "",
  testId = "",
  ...passThroughProps
}) => {
  return (
    <Card className={cn("p-4", "flex", className)} testId={testId} {...passThroughProps}>
      <RecapIcon variant="references" className="mr-4" />
      <div className="flex-auto">
        <Heading variant="h4">{references.title}</Heading>
        <Text variant="p">{references.company}</Text>
        {references.email !== "" && <Text variant="p">{references.email}</Text>}
        {references.phoneNumber !== "" && <Text variant="p">{references.phoneNumber}</Text>}
        <ul className="mt-3">
          {references.bulletPoints.map((bulletPoint, index) => (
            <BulletPoint bulletPoint={bulletPoint} key={index} />
          ))}
        </ul>
      </div>
      <div className="w-1/3 text-right">
        <Button id={references._id} type="button" variant="secondary" onClick={onEdit} className="mr-2">
          <Icon variant="editPencil" size="small" className="mr-2" /> Edit
        </Button>
        <Button id={references._id} type="button" variant="danger" onClick={onDelete}>
          <Icon variant="trash" size="small" className="mr-2" />
          Delete
        </Button>
      </div>
    </Card>
  );
};

export default ReferencesRecap;
