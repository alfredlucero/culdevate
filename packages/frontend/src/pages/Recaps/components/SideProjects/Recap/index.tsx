import React from "react";
import cn from "classnames";
import { format } from "date-fns";
import Card from "../../../../../components/Card";
import Text from "../../../../../components/Text";
import Heading from "../../../../../components/Heading";
import Button from "../../../../../components/Button";
import Icon from "../../../../../components/Icon";
import BulletPoint from "../../BulletPoint";
import RecapIcon from "../../RecapIcon";
import { RecapSideProjects } from "../../../../../interfaces/recaps.interface";
import { CommonProps } from "../../../../../components/commonProps";

interface SideProjectsRecapProps extends CommonProps {
  sideProjects: RecapSideProjects;
  onEdit: (e: React.MouseEvent) => void;
  onDelete: (e: React.MouseEvent) => void;
}

const SideProjectsRecap: React.FC<SideProjectsRecapProps> = ({
  sideProjects,
  onEdit,
  onDelete,
  className = "",
  testId = "",
  ...passThroughProps
}) => {
  return (
    <Card className={cn("p-4", "flex", className)} testId={testId} {...passThroughProps}>
      <RecapIcon variant="projects" className="mr-4" />
      <div className="flex-auto">
        <Heading variant="h4">{sideProjects.title}</Heading>
        <Text variant="p">By {sideProjects.creators}</Text>
        <Text variant="p" italic={true} className="text-gray-600">
          {format(sideProjects.startDate, "MMMM yyyy")} -{" "}
          {sideProjects.endDate ? format(sideProjects.endDate, "MMMM yyyy") : "Present"}
        </Text>
        <ul className="mt-3">
          {sideProjects.bulletPoints.map((bulletPoint, index) => (
            <BulletPoint bulletPoint={bulletPoint} key={index} />
          ))}
        </ul>
      </div>
      <div className="w-1/3 text-right">
        <Button id={sideProjects._id} type="button" variant="secondary" onClick={onEdit} className="mr-2">
          <Icon variant="editPencil" size="small" className="mr-2" /> Edit
        </Button>
        <Button id={sideProjects._id} type="button" variant="danger" onClick={onDelete}>
          <Icon variant="trash" size="small" className="mr-2" />
          Delete
        </Button>
      </div>
    </Card>
  );
};

export default SideProjectsRecap;
