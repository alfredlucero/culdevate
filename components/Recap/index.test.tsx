import React from "react";
import { render, cleanup } from "react-testing-library";
import Recap, { RecapProps } from "./index";

afterEach(cleanup);

describe("<Recap />", () => {
  const sampleRecapProps: RecapProps = {
    title: "Sandia National Labs",
    description: "Web Developer Intern",
    type: "work",
    startDate: "2016/06/20",
    endDate: "2017/06/10",
  };

  test("should render the component", () => {
    const { container } = render(<Recap {...sampleRecapProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("should display to the present when there is no end date", () => {
    const sampleRecapPropsWithoutEndDate = {
      ...sampleRecapProps,
      endDate: null,
    };
    const { getByTestId } = render(<Recap {...sampleRecapPropsWithoutEndDate} />);

    expect(getByTestId("recapDateRange").textContent).toContain("- Present");
  });
});
