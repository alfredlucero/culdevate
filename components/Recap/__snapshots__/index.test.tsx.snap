// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recap /> should render the component 1`] = `
<article
  class="sc-bwzfXH jNsnLW"
>
  <figure>
    <div
      class="sc-bdVaJa jQHlEe"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-suitcase fa-w-16 "
        data-icon="suitcase"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 480h256V80c0-26.5-21.5-48-48-48H176c-26.5 0-48 21.5-48 48v400zm64-384h128v32H192V96zm320 80v256c0 26.5-21.5 48-48 48h-48V128h48c26.5 0 48 21.5 48 48zM96 480H48c-26.5 0-48-21.5-48-48V176c0-26.5 21.5-48 48-48h48v352z"
          fill="currentColor"
        />
      </svg>
    </div>
  </figure>
  <section
    class="sc-htpNat cyYjkk"
  >
    <header
      class="sc-bxivhb ePMNDB"
    >
      <h2>
        Sandia National Labs
      </h2>
      <h3>
        <time
          data-testid="recapDateRange"
        >
          <em>
            2016/06/20
             - 
            2017/06/10
          </em>
        </time>
      </h3>
    </header>
    <p>
      Web Developer Intern
    </p>
    <footer>
      Edit
    </footer>
  </section>
</article>
`;
